### ISSUE TITLE
[False-Positive] Callable parameter usage violates definition with is_subclass_of()

### ISSUE THREAD

**Issue by andreasschroth** [kalessil/phpinspectionsea#1069]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | False-Positive                          |
| Plugin         | Php Inspections (EA Ultimate), 2.0.8 |
| Language level | PHP 7.2                        |

Current behaviour:
---

Here is some example code:

```php
interface AccountInterface extends EntityInterface
{
}

interface FileInterface extends EntityInterface
{
}

function extract(EntityInterface $object): array
{
    if (!is_subclass_of($object, AccountInterface::class)) {
        throw new InvalidInputException(
            'The passed object does not implement the interface AccountInterface.'
        );
    }
}
```

The inspection "Callable parameter usage violates definition" says here: "Makes no sense, because of parameter type declaration.". But this check definitely makes sense - it checks if the passed object belongs to the interface.

So let's say the passed `$object` is of the type `AccountInterface` - it fulfills the method signature (because `AccountInterface` inherits from `EntityInterface`), but it might also be of the type `FileInterface` which also extends from `EntityInterface` (and thus fulfills the check in the method signature), but does not pass the check inside the method.

Expected behaviour:
---

Do not warn in this case, as the check is useful.

**Comment by kalessil** [comment]
Confirmed, I'll take care about fixing it.

**Comment by kalessil** [comment]
Duplicates #1046 (which if fixed)
